### DISCUSSION TITLE
Question about the bulk data

### DISCUSSION THREAD

**Discussion by PeterGriffinJin** [freelawproject/courtlistener#3277] [Q&A]
Hi there,

Thank you so much for the great work!

1) Can I ask what do the "described_opinion_id", "describing_opinion_id" and "group_id" in the bulk data parentheticals-2023-08-31.csv refer to? Do all of them refer to "id" in the opinion-clusters-2023-08-31.csv?

2) Can I ask what do the "cited_opinion_id" and "citing_opinion_id" in the bulk data citation-map-2023-08-31.csv refer to? Do both of them refer to "id" in the opinion-clusters-2023-08-31.csv as well?

3) What does the "depth" mean in citation-map-2023-08-31.csv? Is it true that the bigger the depth value is, the more relevant the two opinions are?

4) Does different opinions in opinion-clusters-2023-08-31.csv refer to different dockets in dockets-2023-08-31.csv?

5) Does each opinion in opinion-clusters-2023-08-31.csv refer to one case?

Many thanks! I'm looking forward to your reply.

Best,
Bowen

**Comment by mlissner** [comment]
>     1. Can I ask what do the "described_opinion_id", "describing_opinion_id" and "group_id" in the bulk data parentheticals-2023-08-31.csv refer to? Do all of them refer to "id" in the opinion-clusters-2023-08-31.csv?

No, those refer to opinion IDs, not cluster IDs.

>     2. Can I ask what do the "cited_opinion_id" and "citing_opinion_id" in the bulk data citation-map-2023-08-31.csv refer to? Do both of them refer to "id" in the opinion-clusters-2023-08-31.csv as well?

Same.

>     3. What does the "depth" mean in citation-map-2023-08-31.csv? Is it true that the bigger the depth value is, the more relevant the two opinions are?

The `depth` field refers to how many times one opinion cites another. So if it cites it once, that indicates one thing, but if it cites it a dozen times, you know it's really important to the opinion making the citations.

>     4. Does different opinions in opinion-clusters-2023-08-31.csv refer to different dockets in dockets-2023-08-31.csv?

Sorry, can you rephrase this? 

>     5. Does each opinion in opinion-clusters-2023-08-31.csv refer to one case?

Legal terms are pretty poorly defined. I think of a "case" as two parties duking it out in court. They might start at a lower court, then appeal, go back to the lower court, etc. All of that, to me, is a case. We don't have "cases" in our database. 

Below cases are dockets. These are collections of documents at a particular court, arguing about a case. On the docket, there may be many opinions. When there is a panel of judges, you might have several opinions that are issued together, including a lead opinion, a dissent, etc. There's no official word for that, but we call that a cluster of opinions.

So...with that said, I think the answer to your question is "no". We can have multiple opinions from a case, including dissents, concurrences, different opinions on different days and at different courts.
 
> Many thanks! I'm looking forward to your reply.

I hope this helps!

**Comment by PeterGriffinJin** [comment]
Hi Mike,

Thank you so much for your quick response!

Can I ask some follow up questions (some just to confirm):
1. Do "described_opinion_id", "describing_opinion_id" and "group_id" in the bulk data parentheticals-2023-08-31.csv refer to "id" entry in the opinions-2023-08-31.csv file? What does "group_id" here represent?

2. Do "cited_opinion_id" and "citing_opinion_id" in the bulk data citation-map-2023-08-31.csv refer to "id" entry in the opinions-2023-08-31.csv file?

3. Do "cluster_id" in the opinions-2023-08-31.csv refer to "id" entry in opinion-clusters-2023-08-31.csv file?

4. Do "docket_id" in the opinion-clusters-2023-08-31.csv refer to "id" entry in dockets-2023-08-31.csv file?

5. Is there any citation relation between opinion-clusters?

Thank you so much for your kind reply and great work!

Best,
Bowen

**Reply by mlissner** [reply]
>  Do "described_opinion_id", "describing_opinion_id" and "group_id" in the bulk data parentheticals-2023-08-31.csv refer to "id" entry in the opinions-2023-08-31.csv file? 

Yes.

> What does "group_id" here represent?

We use a few clever tricks to group highly-similar parentheticals into groups, but it looks like we don't export that data right now. If you wanted to get that table exported, we could do so in the next export. I'd welcome a PR to fix that, but sorry we don't do it now. An oversight.

>  Do "cited_opinion_id" and "citing_opinion_id" in the bulk data citation-map-2023-08-31.csv refer to "id" entry in the opinions-2023-08-31.csv file?

Yes.

>  Do "cluster_id" in the opinions-2023-08-31.csv refer to "id" entry in opinion-clusters-2023-08-31.csv file?

Yes.

>  Do "docket_id" in the opinion-clusters-2023-08-31.csv refer to "id" entry in dockets-2023-08-31.csv file?

Yes.

>  Is there any citation relation between opinion-clusters?

No. Citations are to opinions, not clusters. I know this makes analysis harder, but we think it's the most granular approach and therefore the right way to do it.

**Comment by PeterGriffinJin** [comment]
Hi Mike, 

I appreciate your comprehensive explanation; it has significantly clarified things for me. 

1. Could you please confirm if an "opinion cluster" (as represented in a single row within 'opinion-clusters-2023-08-31.csv') corresponds directly to a "docket" (indicated in one row of 'dockets-2023-08-31.csv') in the bulk data you provided? 

2. Could you please confirm if an "opinion cluster" (as represented in a single row within 'opinion-clusters-2023-08-31.csv') is made up of several "opinion"s (indicated in one row of 'opinion-2023-08-31.csv') in the bulk data you provided? 

3. “We can have multiple opinions from a case, including dissents, concurrences, different opinions on different days and at different courts.” Does that mean each opinion cluster will contain all the opinions from all the judges on the current court including dissents, concurrences? Thus, one opinion cluster is qual to on docket?

4. What do plain_text, html, html_lawbox, html_columbia, html_anon_2020, xml_harvard, html_with_citations in an opinion refer to? Which of them are refer to the same thing?

5. How can we know which opinion is written by which judge? It seems that opinion clusters are given the judges information, but the opinions are not?

6. What does "date_created" and "date_modified" mean? Are they referring to the created and modified data in your database?


Thank you for assisting with this clarification. I really appreciate that!

Best,
Bowen

**Reply by mlissner** [reply]
> 1. Could you please confirm if an "opinion cluster" (as represented in a single row within 'opinion-clusters-2023-08-31.csv') corresponds directly to a "docket" (indicated in one row of 'dockets-2023-08-31.csv') in the bulk data you provided?

It's a relational database, so yes, the docket_id you see in each cluster corresponds to the row in the docket table.

> 2. Could you please confirm if an "opinion cluster" (as represented in a single row within 'opinion-clusters-2023-08-31.csv') is made up of several "opinion"s (indicated in one row of '[opinion-2023](https://www.courtlistener.com/opinion/2023/github-slug/)-08-31.csv') in the bulk data you provided?

Yes, one or more opinions can be in a cluster.

> 3. “We can have multiple opinions from a case, including dissents, concurrences, different opinions on different days and at different courts.” Does that mean each opinion cluster will contain all the opinions from all the judges on the current court including dissents, concurrences? Thus, one opinion cluster is qual to on docket?

That's the goal, but in practice we'll often have a "combined" opinion that has the dissent, concurrence, etc all in a single opinion object, and sometimes we don't have the judge information.

> 4. What do plain_text, html, html_lawbox, html_columbia, html_anon_2020, xml_harvard, html_with_citations in an opinion refer to? Which of them are refer to the same thing?

Please see the documentation, and let me know if you still have questions.

> 5. How can we know which opinion is written by which judge? It seems that opinion clusters are given the judges information, but the opinions are not?

Opinions have an author field.

> 6. What does "date_created" and "date_modified" mean? Are they referring to the created and modified data in your database?

Yes.

**Reply by PeterGriffinJin** [reply]
Hi Mike,

Thank you so much for your detailed response!

"What do plain_text, html, html_lawbox, html_columbia, html_anon_2020, xml_harvard, html_with_citations in an opinion refer to? Which of them are refer to the same thing? -> Please see the documentation, and let me know if you still have questions."

Can I ask where I can find the documentation for this part? 

Thank you so much!

**Reply by mlissner** [reply]
https://www.courtlistener.com/help/api/rest/#opinion-endpoint

**Reply by PeterGriffinJin** [reply]
Thanks Mike!

1) So it seems to me that plain_text, html, html_lawbox, html_columbia, html_anon_2020, html_with_citations all refer to the same item, which is the content of the opinion, it that correct?

2) What is the meaning of xml_harvard by the way?

**Reply by mlissner** [reply]
1. Correct.
2. That's the XML that we got from https://case.law.

**Reply by PeterGriffinJin** [reply]
Hi Mike,

Can I ask which entry gives the real process date of an opinion and docket?

**Reply by mlissner** [reply]
If you're looking for the date of the decision, you'll want the cluster.date_filed field. If you want the date of the case, you'll want the docket.date_filed field. I'm not sure what you mean by "real process" though, sorry.
